### ISSUE TITLE
Gravity

### ISSUE THREAD

**Issue by sidhu1012** [sympy/sympy#20028]
ref issue #20020

@moorepants, I had this in mind

I was thinking that mechanics\particle.py should have a function to calculate gravitational force of a particle w.r.t to another particle so that if user wishes to calculate for just 2 particle system he could do it directly (scalar value).
And have a similar function in class particle to calculate gravitational potential of the particle at a specified distance.

**Comment by moorepants** [comment]
Yes, something like that would be useful. We'd need to support it for any given set of particles and rigid bodies. It is a bit tricky getting it right for rigidbodies. There is an example in Kane's book that shows how to calculate the rigid body effects.

**Comment by sidhu1012** [comment]
> Yes, something like that would be useful. We'd need to support it for any given set of particles and rigid bodies. It is a bit tricky getting it right for rigidbodies. There is an example in Kane's book that shows how to calculate the rigid body effects.

Shall I work on particle first?
And begin with with 2 body system? 
Gravitational force and gravitational potential

**Comment by thambi999** [comment]
Plz what yr whan

Sent from Yahoo Mail on Android 
 
  On Tue, Sep 1, 2020 at 16:37, sidhu1012<notifications@github.com> wrote:   



Yes, something like that would be useful. We'd need to support it for any given set of particles and rigid bodies. It is a bit tricky getting it right for rigidbodies. There is an example in Kane's book that shows how to calculate the rigid body effects.


Shall I work on particle first?
And begin with with 2 body system?
Gravitational force and gravitational potential

**Comment by sidhu1012** [comment]
I have been searching for a while but couldn't find a function in sympy to convert a vector into unit vector and a function which returns magnitude of vector.
Are those functions available ? If yes then please specify them along with their module path.

**Comment by sidhu1012** [comment]
Does normalize function of class vector returns Unit vector of that vector?

**Comment by moorepants** [comment]
> Does normalize function of class vector returns Unit vector of that vector?

Yes.

**Comment by moorepants** [comment]
> Shall I work on particle first?
> And begin with with 2 body system?

I recommend adding an example to the documentation showing how to form the equations for a 2-particle system in 3D. After that you could work on more advanced examples with more bodies and particles. Then after all of that, you can try to design a function that does this in an automated way.

**Comment by sidhu1012** [comment]
> > Shall I work on particle first?
> > And begin with with 2 body system?
> 
> I recommend adding an example to the documentation showing how to form the equations for a 2-particle system in 3D. After that you could work on more advanced examples with more bodies and particles. Then after all of that, you can try to design a function that does this in an automated way.

Okay

**Comment by sidhu1012** [comment]
I' m facing an issue.
When I write :


from sympy import symbols
from sympy import symbols
from sympy.physics.mechanics import Point, Particle, ReferenceFrame


m1, m2, r1, r2 = symbols('m1 m2 r1 r2')
N = ReferenceFrame('N')
O1 = Point('O1')
P1 = Particle('P1', O1, m1)
O2 = Point('O2')
O2.set_pos(O1, 10 * N.x)
P2 = Particle('P2', O2, m2)

vect_dist = P1.point.pos_from(P2.point)
r = vect_dist.magnitude() ** 2
r_normalize = vect_dist.normalize()


Now when I do some operations:

print(P1.mass * r)
output : 100*m1

print(P1.mass * r_normalize)
output: - m1*N.x

print(P1.mass * r * r_normalize)
output: - 100*m1*N.x

Till now all outputs are as expected. But on running this

print((P1.mass * r_normalize)/r)
output: - m1/100*N.x

expected_output: -m1 * N1/100

Unit vector is going in denominator when multiplied in numerator on dividing by magnitude.
Is this the expected behavior or I'm missing somethin

**Comment by sidhu1012** [comment]
@moorepants Can you please help me out with issue #20039 , as this the last hurdle in my code for this issue.

**Comment by Harsh-Upadhayay** [comment]
Is this issue still open or is it resolved?

**Comment by moorepants** [comment]
The issue is still open.
